diff --git a/src/util.cpp b/src/util.cpp
index 4659ff7..b63edd0 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -531,7 +531,7 @@ fs::path GetDefaultDataDir()
     return GetSpecialFolderPath(CSIDL_APPDATA) / "ioncoin";
 #else
     fs::path pathRet;
-    char* pszHome = getenv("HOME");
+    char* pszHome = getenv("SNAP_USER_COMMON");
     if (pszHome == nullptr || strlen(pszHome) == 0)
         pathRet = fs::path("/");
     else
